src/datadog/tracer_config.h

```diff
@@ -127,19 +148,41 @@ struct TracerConfig {
 
   // `integration_name` is the name of the product integrating this library.
   // Example: "nginx", "envoy" or "istio".
-  std::string integration_name;
+  Optional<std::string> integration_name;
   // `integration_version` is the version of the product integrating this
   // library.
   // Example: "1.2.3", "6c44da20", "2020.02.13"
-  std::string integration_version;
+  Optional<std::string> integration_version;
+};
+
+struct TracerDefaultConfig final {
+  bool log_on_startup = true;
+  bool report_traces = true;
+  bool report_hostname = false;
+  bool report_telemetry = true;
+  bool delegate_trace_sampling = false;
+  bool generate_128bit_trace_ids = true;
+  std::string service_type = "web";
+  std::size_t max_tags_header_size = 512;
+  std::vector<PropagationStyle> injection_styles = {PropagationStyle::DATADOG,
```

Comment:
Proposal: make `PropagationStyle` an arithmetic enum. Usage example:
````cpp
struct TracerDefaultConfig final {
   PropagationStyle injection_styles = PropagationStyle::DATADOG | PropagationStyle::W3C;
   PropagationStyle extraction_styles = PropagationStyle::DATADOG | PropagationStyle::W3C;
};

...

if (extraction_styles & PropagationStyle::W3C) {
   ....
}
````

Why?
- vector is overkill and tedious to use
- same information packed in a byte.